astropy/time/core.py

```diff
@@ -803,12 +803,15 @@ def _apply(self, method=None, *args, **kwargs):
             # Apply the method to any value arrays (though skip if there is only
             # a single element and the method would return a view, since in
             # that case nothing would change).
-            if method is not None and val is not None:
-                if method == 'copy' or method == 'flatten' and val.size == 1:
+            if method is not None and val is not None and hasattr(val, method):
```

Comment:
I'm not sure if it helps the code clarity, but I think it is possible to write this simply as `if hasattr(val, method):`, since:
- `None` object cannot be an attribute (`hasattr(val, None)` is always False)
- `None` object has no public attributes (`hasattr(None, method)` is always False as long as `method` is not a dunder-string.